src/Microsoft.HttpRepl.IntegrationTests/Commands/SetSwaggerCommandTests.cs

```diff
@@ -0,0 +1,266 @@
+            List<string> directoryNames = directoryStructure.DirectoryNames.ToList();
+            string expectedDirectoryName = "api";
+
+            Assert.Single(directoryNames);
+            Assert.Equal("api", directoryNames.First());
+
+            IDirectoryStructure childDirectoryStructure = directoryStructure.GetChildDirectory(expectedDirectoryName);
+
+            Assert.Empty(childDirectoryStructure.DirectoryNames);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithChildDirectories_SetsHttpStateSwaggerStructureWithChildDirectorStuctureInfo()
+        {
+            string response = @"{
+  ""swagger"": ""2.0"",
+  ""paths"": {
+    ""/api/Employees"": {
+      ""get"": {
+        ""tags"": [ ""Employees"" ],
+        ""operationId"": ""GetEmployee"",
+        ""consumes"": [],
+        ""produces"": [ ""text/plain"", ""application/json"", ""text/json"" ],
+        ""parameters"": [],
+        ""responses"": {
+          ""200"": {
+            ""description"": ""Success"",
+            ""schema"": {
+              ""uniqueItems"": false,
+              ""type"": ""array""
+            }
+          }
+        }
+      }
+    },
+    ""/api/Values"": {
+      ""post"": {
+        ""tags"": [ ""Values"" ],
+        ""operationId"": ""Post"",
+        ""consumes"": [ ""application/json-patch+json"", ""application/json"", ""text/json"", ""application/*+json"" ],
+        ""produces"": [],
+        ""parameters"": [
+          {
+            ""name"": ""value"",
+            ""in"": ""body"",
+            ""required"": false,
+            ""schema"": { ""type"": ""string"" }
+          }
+        ],
+        ""responses"": { ""200"": { ""description"": ""Success"" } }
+      }
+    }
+  }
+}";
+            MockedShellState shellState = new MockedShellState();
+            string parseResultSections = "set swagger http://localhost:5050/somePath";
+            IDirectoryStructure directoryStructure = await GetDirectoryStructure(shellState, response, parseResultSections).ConfigureAwait(false);
+            List<string> directoryNames = directoryStructure.DirectoryNames.ToList();
+            string expectedDirectoryName = "api";
+
+            Assert.Single(directoryNames);
+            Assert.Equal(expectedDirectoryName, directoryNames.First());
+
+            IDirectoryStructure childDirectoryStructure = directoryStructure.GetChildDirectory(expectedDirectoryName);
+            List<string> childDirectoryNames = childDirectoryStructure.DirectoryNames.ToList();
+
+            Assert.Equal(2, childDirectoryNames.Count);
+            Assert.Equal("Employees", childDirectoryNames.First());
+            Assert.Equal("Values", childDirectoryNames.ElementAt(1));
+        }
+
+        private async Task<IDirectoryStructure> GetDirectoryStructure(MockedShellState shellState, string response, string parseResultSections)
```

Comment:
FYI, this will get moved to ICommandTestHelper once these changes are merged and when I send out pr for SetBaseCommandTests